plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java

```diff
@@ -59,7 +66,25 @@ private void showLogs(IPod pod, ExecutionEvent event) {
 			MessageDialog.openError(HandlerUtil.getActiveShell(event), "Logs Unavailable", NLS.bind("The log is unavailable while the pod is in {0} state.", pod.getStatus()));
 			return;
 		}
-		new PodLogsJob(pod).schedule();
+		Collection<IContainer> containers = pod.getContainers();
+		String containerName = null;
+		if(containers.size() > 1) {
+			List<String> names = containers.stream().map(c->c.getName()).collect(Collectors.toList());
+			Collections.sort(names);
+			ElementListSelectionDialog dialog = new ElementListSelectionDialog(HandlerUtil.getActiveShell(event), new LabelProvider());
+			dialog.setElements(names.toArray());
+			dialog.setTitle("Pod Containers");
+			dialog.setMessage("Select a pod container");
+			dialog.setMultipleSelection(false);
+			int result = dialog.open();
+			if(Window.CANCEL == result) {
+				return;
+			}
+			containerName = (String) dialog.getFirstResult();
+		}else if (containers.size() == 1){
+			containerName = containers.iterator().next().getName();
+		}
+		new PodLogsJob(pod, containerName).schedule();
```

Comment:
technically if containers.size() == 0, then containerName == null, so no need to schedule the job